pkgs/development/tools/build-managers/bazel/bazel_7/bazel-repository-cache.nix

```diff
@@ -94,15 +132,21 @@ let
       addSources = acc: withToplevelValue (withRemotePatches (withArchives acc));
     in
     if builtins.isAttrs value && value ? attributes
-      && builtins.isAttrs attrs && attrs ? name
+      && value ? ruleClassName
+      && (value.ruleClassName == "http_file"
+      || value.ruleClassName == "http_archive"
+      || value.ruleClassName == "http_jar"
+      || value.ruleClassName == "_distdir_tar"
+      || value.ruleClassName == "python_repository")
+      && builtins.isAttrs attrs
       && (attrs ? sha256 || attrs ? integrity)
       && (attrs ? urls || attrs ? url)
-      && f attrs.name
+      && f name
     then addSources acc
     else acc;
 
   requiredSourcePredicate = n: requiredDepNamePredicate (sanitize n);
-  requiredDeps = foldlJSON (extract_source requiredSourcePredicate) { } modules;
+  requiredDeps = foldlModuleDepGraph (extract_source requiredSourcePredicate) { } modules // foldlGeneratedRepoSpecs (extract_source requiredSourcePredicate) { } modules;
```

Comment:
I'm not sure about the naming "fold..." here. Also, this looks a bit verbose.

Reply:
Yes, parsing twice is not ideal, but parsing at all may not survive 7.2.0, so good enough for now.